winton_kafka_streams/processor/_stream_thread.py

```diff
@@ -205,12 +271,25 @@ def run(self):
             self.commitAll()
             self.shutdown()
 
-    def processAndPunctuate(self, record):
-        task = self.tasks[record.partition()]
-        task.add_records([record])
-        task.process()
-        if task.needCommit:
-            self.commit(task)
+    def add_records_to_tasks(self, records):
+        for record in records:
+            self.tasks[record.partition()].add_records([record])
+
+    def process_and_punctuate(self):
```

Comment:
Mitigation #1: Keep processing records until we've emptied the inbound queues, so we don't commit after every single individual message.